### PR TITLE
Added additional ConfigContext implementations

### PR DESCRIPTION
**Pull request by dekobon**
This PR adds the following:
* Java system properties based configuration
* Map based configuration
* MANTA_TIMEOUT environment variable
* Readme improvements
* Version bump to 1.6.0

### CODE REVIEW

**Comment by dekobon** [comment]
@phillipross If you get a moment, could you please review this PR?

File: pom.xml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
   <groupId>com.joyent.manta</groupId>
   <artifactId>java-manta</artifactId>
-  <version>1.5.4</version>
+  <version>1.6.0</version>
```
Comment (phillipross) [inline pom.xml]: We should be using SNAPSHOT versions and then change to a non-snapshot version during the release process.
ie... 1.6.0 should be 1.6.0-SNAPSHOT until release time.

**Comment by phillipross** [comment]
@dekobon Other than the version change to a SNAPSHOT version, everything looks fine.

**Comment by dekobon** [comment]
Thanks for the input. I'll change it. I've been wondering how to properly do snapshots for sometime. I thought I knew how and then I started to use gradle. Then it got me really confused. :|

**Pull request merged**